Dockerfile

```diff
@@ -0,0 +1,31 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# User used targeting base image
+ARG BASE_GO_IMAGE
+# Build the agent base image
+ARG BASE_BUILDER_IMAGE='golang:1.18'
+
+FROM ${BASE_BUILDER_IMAGE} as builder
+# Go Agent Version
+ARG VERSION
+WORKDIR /skywalking-go
+COPY . .
+RUN VERSION=$VERSION make -C tools/go-agent linux
+
+FROM ${BASE_GO_IMAGE}
```

Comment:
I am missing a couple of things here:
1. User
2. labels for referencing the repository e.g. opencontainers labels.
3. More documentation about what a BASE_GO_IMAGE is and where is it supposed to have the go bin or mimimun version.

Reply:
> User

Currently, it will be used in the skywalking showcase to prevent manual downloading of the latest project and repeating the build process.

> labels for referencing the repository e.g. opencontainers labels.

Do you mean tags? It has been defined [in the Makefile](https://github.com/apache/skywalking-go/pull/82/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R124).

> More documentation about what a BASE_GO_IMAGE is and where is it supposed to have the go bin or mimimun version.

For now, the documentation is more to declare how to use it. The support go version list has been [documented there](https://github.com/apache/skywalking-go/pull/82/files#diff-587e31dd45aa2559ac2b70b2174858a872f7e5565036cfdbf6dceb4afe9bbfc6R26).

Reply:
I meant this set of annotations https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys